packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -126,6 +155,33 @@ export function useGetRestrictionUiType(
   return RestrictionUiType.AccessBlockedToAnonymous
 }
 
+/**
+ * This hook determines if
+ * - the entity is a FileEntity, AND
+ * - the caller has permission to fetch the dataFileHandle, AND
+ * - the dataFileHandle is an instance of ExternalFileHandleInterface (i.e. the file is not controlled by Synapse)
+ * Note that this requires an additional API call that cannot be batched, so it should be avoided in bulk contexts if possible.
+ */
+function useGetIsExternalFileHandle(
+  entityId: string,
+  options: { enabled: boolean },
```

Comment:
ideally use the full type if you can
```suggestion
  options: Partial<UseQueryOptions<EntityBundle, SynapseClientError, boolean>>,
```